relay-server/src/actors/project_cache.rs

```
@@ -519,30 +646,121 @@ impl ProjectCacheBroker {
         &mut self,
         message: CheckEnvelope,
     ) -> Result<CheckedEnvelope, DiscardReason> {
-        let project = self.get_or_create_project(message.envelope.meta().public_key());
-
+        let CheckEnvelope { envelope, context } = message;
+        let project = self.get_or_create_project(envelope.meta().public_key());
         // Preload the project cache so that it arrives a little earlier in processing. However,
         // do not pass `no_cache`. In case the project is rate limited, we do not want to force
         // a full reload. Fetching must not block the store request.
         project.prefetch(false);
-
-        project.check_envelope(message.envelope, message.context)
+        project.check_envelope(envelope, context)
     }
 
-    fn handle_validate_envelope(&mut self, message: ValidateEnvelope) {
-        // Preload the project cache for dynamic sampling in parallel to the main one.
-        if let Some(sampling_key) = utils::get_sampling_key(&message.envelope) {
-            self.get_or_create_project(sampling_key)
-                .prefetch(message.envelope.meta().no_cache());
+    /// Handles the processing of the provided envelope.
+    fn handle_processing(
+        &mut self,
+        project_key: ProjectKey,
+        state: Arc<ProjectState>,
+        sampling_state: Option<Arc<ProjectState>>,
+        envelope: Box<Envelope>,
+        envelope_context: EnvelopeContext,
+    ) {
+        if let Ok(CheckedEnvelope {
+            envelope: Some((envelope, envelope_context)),
+            ..
+        }) = self
```

Is it OK to swallow the error case here?

---

That's how this was done before, but  we also can rethink this behaviour now.